### PR TITLE
Ekrengel/add new features to integration routes

### PR DESCRIPTION
**Pull request by ekrengel**

### CODE REVIEW

File: nodejs/awsx/apigateway/api.ts [hunk 1]
```diff
@@ -155,6 +155,32 @@ function isStaticRoute(route: Route): route is StaticRoute {
...
+     * the API level and by default, the source will be the HEADER.
+     */
+    apiKeyRequired?: boolean;
+
+    /**
+     * Authorizers allows you to define Lambda authorizers be applied for authorization when the
+     * the route is called. The authorizer will get applied to all static resources defined by the
+     * localPath.
+     */
+    authorizers?: Authorizer[] | Authorizer;
```
Comment (CyrusNajmabadi) [inline nodejs/awsx/apigateway/api.ts]: these four now seem common to most routes (except 'raw json') right?  If so, can we extract out to `interface BaseRoute` and then have the others extend from that so we don't have to copy this into all of them?

File: nodejs/awsx/apigateway/api.ts [hunk 2]
```diff
@@ -986,11 +1012,31 @@ function addIntegrationRouteToSwaggerSpec(
...
+    }
+    if (route.requestValidator) {
+        swaggerOpWithoutProxyPathParam["x-amazon-apigateway-request-validator"] = route.requestValidator;
+        swaggerOpWithProxyPathParam["x-amazon-apigateway-request-validator"] = route.requestValidator;
+    }
+    if (route.apiKeyRequired) {
+        addAPIkeyToSecurityDefinitions(swagger);
+        addAPIKeyToSwaggerOperation(swaggerOpWithoutProxyPathParam);
+        addAPIKeyToSwaggerOperation(swaggerOpWithProxyPathParam);
+    }
```
Comment (CyrusNajmabadi) [inline nodejs/awsx/apigateway/api.ts]: not possible to extract a function that will do the 4 steps, but then call it for both swaggerOpWithoutProxyPathParam and swaggerOpWithProxyPathParam?

Comment (CyrusNajmabadi) [reply nodejs/awsx/apigateway/api.ts]: feels like we've triplicated this with addEventHandlerRouteToSwaggerSpec and addStaticRouteToSwaggerSpec (Esp for the file/dir handling).  This feels like this must be extractable.

File: nodejs/awsx/aws_test.go [hunk 3]
```diff
@@ -489,23 +507,23 @@ func validateAPITests(apiTests []apiTest) func(t *testing.T, stack integration.R
 
 			if tt.requiredToken != nil {
 				resp := GetHTTP(t, req, 401)
-				assertRequestBody(t, `{"message":"Unauthorized"}`, resp)
+				assertRequestBody(t, `{"message":"Unauthorized"}`, false, resp)
```
Comment (CyrusNajmabadi) [inline nodejs/awsx/aws_test.go]: can you do `false /*skipBodyValidation*/`?  Passing constants like this makes it hard to tell at his callsite what's going on.

File: nodejs/awsx/apigateway/api.ts (hunk 1 above)
Comment (ekrengel) [reply nodejs/awsx/apigateway/api.ts]: Its not in the Raw Data Route. I'll spend some time thinking about how to add it to that..

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,7 @@
 ## 0.18.3 (Unreleased)
 
+- Add Typescript support for Authorizers, API Keys and Request Validation to Integration Routes in API Gateway
```
Comment (CyrusNajmabadi) [inline CHANGELOG.md]: remove 'TypeScript'.

**Pull request merged**